components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/TenantSyncListener.java

```diff
@@ -0,0 +1,431 @@
+    }
+
+    @Override
+    public void onTenantInitialActivation(int tenantId) throws StratosException {
+        // Not implemented.
+    }
+
+    @Override
+    public void onTenantActivation(int tenantId) throws StratosException {
+
+        if (canCreateTenant(tenantId)) {
+            TenantInfoBean tenantInfo = new TenantInfoBean();
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+                String domain = realmService.getTenantManager().getTenant(tenantId).getDomain();
+
+                tenantInfo.setTenantDomain(domain);
+                tenantInfo.setTenantId(tenantId);
+                tenantInfo.setActive(true);
+                sendEvent(tenantInfo, TenantManagement.ACTION_ACTIVATE, TenantManagement.EVENT_ACTIVATE_TENANT_URI);
+            } catch (UserStoreException e) {
+                LOG.error("Error while activating tenant ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+    }
+
+    @Override
+    public void onTenantDeactivation(int tenantId) throws StratosException {
+
+        if (canCreateTenant(tenantId)) {
+            TenantInfoBean tenantInfo = new TenantInfoBean();
+            try {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+                RealmService realmService = IdentityMgtServiceComponent.getRealmService();
+                String domain = realmService.getTenantManager().getTenant(tenantId).getDomain();
+
+                tenantInfo.setTenantDomain(domain);
+                tenantInfo.setTenantId(tenantId);
+                tenantInfo.setActive(false);
+                sendEvent(tenantInfo, TenantManagement.ACTION_DEACTIVATE, TenantManagement.EVENT_ACTIVATE_TENANT_URI);
+            } catch (UserStoreException e) {
+                LOG.error("Error while deactivating tenant ", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+    }
+
+    @Override
+    public void onSubscriptionPlanChange(int tenentId, String oldPlan, String newPlan) throws StratosException {
+        // Not implemented.
+    }
+
+    @Override
+    public int getListenerOrder() {
+
+        return 0;
+    }
+
+    @Override
+    public void onPreDelete(int tenantId) throws StratosException {
+        // Not implemented.
+    }
+
+    /**
+     * Check whether tenant sharing is enabled and whether the creation is only a root organization.
+     *
+     * @param tenantId id of the tenant
+     * @return boolean Tenant creation firing is enable or not.
+     */
+    private boolean canCreateTenant(int tenantId) {
```

Comment:
Suggestion: isTenantEventFiringEnabled or isTenantEventsEnabled

Reply:
fixed in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/d32d019d0cf128847ec73bc433a30bb8d0ec34ee